charts/spinnaker/templates/configmap/halyard-config.yaml

```diff
@@ -132,6 +132,10 @@ data:
                 {{ if $account.readPermissions }}--read-permissions {{ join " --read-permissions " $account.readPermissions }}{{ end }} \
                 {{ if $account.writePermissions }}--write-permissions {{ join " --write-permissions " $account.writePermissions }}{{ end }} \
                 --provider-version v2
+    {{- range $index, $customResource := .Values.kubeConfig.customResources }}
+    $HAL_COMMAND config provider kubernetes account $PROVIDER_COMMAND {{ $name }} --docker-registries dockerhub \
+                --context {{ $account.context }} --provider-version v2 --add-custom-resource={{ $customResource }} --spinnaker-kind=serverGroupManager --no-validate
```

Comment:
Could we merge this command with the one above by doing the same `{{ if .Values.kubeConfig.customResources }} <then add all the flags> {{ end }}` as we do for other fields too? As otherwise this would change it from 1 command per account to 4, which impacts the `install-using-hal` job duration significantly.

Reply:
Let me check. Before your comment, I assumed that the resources could be added one by one only, but I'll check if it is possible to use multiple `--add-custom-resource=` within one command

Reply:
Hmm I see. The [code](https://github.com/spinnaker/halyard/blob/507e1d4d7eb7cab7b85586651a2b6662470482e7/halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/config/providers/kubernetes/KubernetesEditAccountCommand.java#L162) also shows it is just a `String` instead of a `Set<String>`.

We should see before going with this approach whether it is feasible from an applying POV. If this 2x or 3x the apply duration, it would be really slowing us down. Perhaps at that point we should look into whether we can mount this part of the config instead of using `hal` commands.

And there was no generic/global place to configure this right? It _had_ to be on the Kubernetes account definition?

Reply:
It might be a repetition here, but AFAIU the time spent comes from (double?) validation. Here, the `--no-validate` flag is used. Shouldn't that partly mitigate the issue?

Just to be sure I understand here, this will be called `numberCRDs * numberAccounts` times , where in an ideal state `numberAccounts = numberTeams * numberClusters`, is that correct?

Reply:
>  where in an ideal state `numberAccounts = numberTeams * numberClusters`, is that correct?

As for an OSS Helm chart, that is up to the user. But in our case, this is `numberAccounts = numberNamespaces * numberClusters`. So this can easily become 300+ calls. I wonder how much the `--no-validate` helps here. I noticed that one `hal config providers edit ...` call takes roughly 1 second; not sure how much of that is validation.

Reply:
Then to move forward on this, maybe we could `time` this? We should be able to extrapolate and hence get an expected time spent on an "ideal" setup. Based on that, we can then make a decision on whether that is acceptable, wdyt?

Reply:
Thanks for your comments @Badbond and @LucaHennart !
I couldn't find another way to provide custom resources than by executing the hal command with the flag multiple times. And the `--no-validate` was indeed added to avoid extra validation time.